pagopa/io-app
# [IOPID-2566] Refactor AuthErrorComponent

## Short description
This PR removes the `onRetry` function from the `AuthErrorScreen` navigation params in order to suppress the warning related to it. The retry logic is now handled in a dedicated redux reducer.
>[!Important]
>In the `ts/screens/authentication/IdpLoginScreen.tsx` component the `navigate` method has been replaced with the `replace`. This change gives us the possibility to skip the `retryLollipopLogin` step because the `replace` unmounts the route and in a further navigation on the `ts/screens/authentication/IdpLoginScreen.tsx` the `useLollipopLoginSource` hook will be executed from start.

## List of changes proposed in this pull request 
- Removed `onRetry` from the `AuthErrorScreen` navigation params
- Added `isNativeLogin` param into the `AuthErrorScreen` navigation params
- Moved the `ts/screens/authentication/IdpLoginScreen.tsx` and `ts/screens/authentication/idpAuthSessionHandler.tsx` retry logic in a dedicated redux reducer
- Replaced the `navigate` function with the `replace` function in the `ts/screens/authentication/IdpLoginScreen.tsx`

## Demo

<details><summary>iOS</summary>

#### Standard login

|Before|After|
|------|-------|
|<video src="https://github.com/user-attachments/assets/35e2c2b7-3514-4db8-a7b1-7d497ab03372"></video>|<video src="https://github.com/user-attachments/assets/57f6ddfe-1a07-4821-b1d3-35ff9ee90c14"></video>|

#### Native login

|Before|After|
|------|-------|
|<video src="https://github.com/user-attachments/assets/5d4d57ed-7915-42ed-86d5-fd88383fcfe5"></video>|<video src="https://github.com/user-attachments/assets/144ad77e-55ce-4df6-bbcd-28cbcf63924a"></video>|

</details>

<details><summary>Android</summary>

#### Standard login
|Before|After|
|------|-------|
|<video src="https://github.com/user-attachments/assets/00e1498b-dad5-4e0b-b40e-345d661a6566"></video>|<video src="https://github.com/user-attachments/assets/c7b210c4-24bf-42e3-934f-2f2f7f76e54f"></video>|

#### Native login
|Before|After|
|------|-------|
|<video src="https://github.com/user-attachments/assets/8405ee01-5c99-4dd2-9774-4698aa953928"></video>|<video src="https://github.com/user-attachments/assets/4c64b181-0d24-47ef-aada-cfba9fee0f39"></video>|

</details>

## How to test
Replicate what is shown in the demos. By default, the standard login flow is enabled. To test the native login, ensure that the local feature flag is enabled in the environment variables and that the remote one is enabled on the dev-server.
